adapter/internal/api/apis_impl.go

```diff
@@ -288,16 +280,6 @@ func ApplyAPIProjectFromAPIM(
 		}
 	}
 
-	// undeploy APIs with other vhosts in the same gateway environment
-	for vhost, environments := range vhostsToRemove {
-		if vhost == "" {
-			// ignore if vhost is empty, since it deletes all vhosts of API
-			continue
-		}
-		if err := xds.DeleteAPIsWithUUID(vhost, apiYaml.ID, environments, apiProject.OrganizationID); err != nil {
-			return deployedRevisionList, err
-		}
-	}
```

Comment:
Don't we need this now? @renuka-fernando

Reply:
This is used in the APIM scenario where we can only set one Vhost for an environment.
For example, deploy petstore API
1. revision 1 -> env: default -> vhost: foo.com
2. revision 2 -> en : default -> vhost: bar.com

So when deploying revision 2, choreo connect has to un-deploy API from foo.com to maintain the API is deployed on one Vhost.

Since we need more Vhosts for one environment, we have to remove this.

Reply:
Also included the info in the commit message.